charts/linkerd2/templates/_config.tpl

```diff
@@ -31,7 +31,7 @@
   },
   "ignoreInboundPorts":[
     {{- $ports := splitList "," .Values.global.proxyInit.ignoreInboundPorts -}}
-    {{- if gt (len $ports) 1}}
+    {{- if (and (gt (len $ports) 0) (ne (first $ports) "")) }}
```

Comment:
What are the different cases that we are expecting and how do they arise?  Are these inputs valid?

* $ports is length 0
* $ports is length 1 and contains just an empty string
* $ports is length 2 and the second element is an empty string

Reply:
So if by inbputs you mean what is passed to `global.proxyInit.ignoreInboundPorts`, then: 

- if its empty, its an empty string, so after split $ports will be a list of one empty string, which we filter
- if its just "", then its the same case as above
- if the length is two, then we do not do any validation to guard against an empty string not being a port.

Essentially what is happenign here is the following: 
- We fix the problem where specifying just one port will result in it not getting written to the config map
- We make sure that we have an empty `global.proxyInit.ignoreInboundPorts` we do not end up with an array that looks like `[""]` in the config map. 

I think any further validation on input is pushing the helm functionality a bit too far. WDYT? 
As you suggested in https://github.com/linkerd/linkerd2/pull/3953#discussion_r369298051, I am fine with factoring all that out into a function of its own so we can reuse it.

Reply:
@zaharidichev Thanks for the explanation. There is a good chance that, in the future, I will forget `{{- if (and (gt (len $ports) 0) (ne (first $ports) "")) }}` can handle all the cases @adleong pointed out.

Will it work if we check whether `Values.global.proxyInit.ignoreInboundPorts` is empty? If it is, just skip the `splitList` and `range` loop:

```
  "ignoreInboundPorts":[
    {{- if gt (len $.Values.gobal.proxyInit.ignoreInboundPorts) 0 -}}
    {{- $ports := splitList "," .Values.global.proxyInit.ignoreInboundPorts -}}
    {{- $last := sub (len $ports) 1 -}}
    {{- range $i,$port := $ports -}}
    {"port":{{$port}}}{{ternary "," "" (ne $i $last)}}
    {{- end -}}
    {{- end -}}
  ],
```

Reply:
Alright, so I rebased against #3953 (so this needs to go in first) and used your suggested approach for both fixes. Now these splitting functions are refactored in _helpers.tpl. Does that look better?